pytorch_lightning/trainer/connectors/loop_connector.py

```diff
@@ -0,0 +1,68 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Optional
+
+import pytorch_lightning as pl
+from pytorch_lightning.loops import EvaluationLoop, FitLoop, PredictionLoop
+from pytorch_lightning.loops.base import Loop
+
+
+class LoopConnector(object):
+
+    def __init__(self, trainer: "pl.Trainer"):
+        self.trainer = trainer
+
+    def on_trainer_init(self, *args, **kwargs):
+        self.trainer.fit_loop = self.on_fit_loop_init(*args, **kwargs)
+        self.trainer.validate_loop = self.on_validate_loop_init(*args, **kwargs)
+        self.trainer.test_loop = self.on_test_loop_init(*args, **kwargs)
+        self.trainer.predict_loop = self.on_predict_loop_init(*args, **kwargs)
```

Comment:
a loop connector could be useful in general, but we said we wanted to avoid Trainer state management outside the trainer. If Trainer owns loops, imo it would be safer to let the Trainer manage its state.

cc @ananthsub @justusschock